lib/views/course_view/components/pinned_course_view.dart

```diff
@@ -0,0 +1,97 @@
+import 'package:flutter/material.dart';
+
+/// PinnedCourseCard
+///
+/// A card widget representing a pinned course. It displays the course details,
+/// including the course title, date, and duration. Users can toggle the pinned state
+/// by clicking the pin icon, and tapping the card navigates to the course content.
+///
+/// Parameters:
+///   [imageName] - The asset path or URL of the course image.
+///   [title] - The title of the pinned course.
+///   [date] - The date or timestamp associated with the course.
+///   [duration] - The duration or length of the course.
+///   [isPinned] - A boolean indicating whether the course is currently pinned.
+///   [onPinToggle] - A callback function triggered when the user toggles the pin icon.
+///   [onTap] - A callback function triggered when the user taps the card to view the course content.
+///
+class PinnedCourseCard extends StatelessWidget {
+  final String imageName;
+  final String title;
+  final String date;
+  final String duration;
+  final bool isPinned;
+  final VoidCallback onPinToggle;
+  final VoidCallback onTap;
+
+  const PinnedCourseCard({
+    super.key,
+    required this.imageName,
+    required this.title,
+    required this.date,
+    required this.duration,
+    required this.isPinned,
+    required this.onTap,
+    required this.onPinToggle,
+  });
```

Comment:
One small suggestion: Maybe replace 
```dart
final String title;
final String date;
final String duration;
```
with e.g.,
```dart
final Course course
```
so that one doens't have to explicitly pass all course parameters when creating a new PinnedCourseCard

Reply:
I replaced course details with a Course object